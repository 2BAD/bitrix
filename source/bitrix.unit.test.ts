import { describe, expect, it, vi } from 'vitest'
import Bitrix from './bitrix.js'
import * as client from './client/index.js'

describe('Bitrix', () => {
  it('should init', () => {
    expect(() => Bitrix('https://test.com', 'test_token')).not.toThrow()
  })

  it('should pass endpoint and access token to the client', () => {
    const SpiedClient = vi.spyOn(client, 'default')

    Bitrix('https://test.com', 'test_token')
    expect(SpiedClient.mock.calls).toMatchSnapshot()
  })

  it('should expose API methods', () => {
    const bitrix = Bitrix('https://test.com', 'test_token')
    expect(bitrix).toMatchSnapshot()
  })
})
