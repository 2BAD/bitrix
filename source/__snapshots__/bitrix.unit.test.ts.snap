// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bitrix should expose API methods 1`] = `
Object {
  "batch": [Function],
  "call": [Function],
  "contacts": Object {
    "create": [Function],
    "get": [Function],
    "list": [Function],
    "update": [Function],
  },
  "deals": Object {
    "create": [Function],
    "get": [Function],
    "list": [Function],
    "update": [Function],
  },
  "leads": Object {
    "create": [Function],
    "get": [Function],
    "list": [Function],
    "update": [Function],
  },
  "list": [Function],
  "statuses": Object {
    "create": [Function],
    "get": [Function],
    "list": [Function],
    "update": [Function],
  },
}
`;

exports[`Bitrix should pass endpoint and access token to the client 1`] = `
Array [
  Array [
    "https://test.com",
    "test_token",
  ],
]
`;
